doc/src/manual/workflow-tips.md

```diff
@@ -26,6 +26,10 @@ line. A common pattern includes the following elements:
 
     end
     ```
+    The key here is to encapsulate the code in a module.
+    If it were not so, and the code was ran directly in the `Main` module of the REPL,
+    you would not be able to remove methods or edit `struct` definitions, without restarting Julia.
+    
```

Comment:
```suggestion
    The key here is to encapsulate the code in a module.
    This makes the global state more isolated, so that it is not affected by prior content in the `Main` module of the REPL, 
    or methods and `struct` definitions added before.
    This reduces the need to restart Julia in this workflow.

```

Reply:
(changed to word this only in the positive, and to mention another benefit of this workflow tip)

Reply:
Thank you for the review and suggestion :)

The isolation from previous REPL commands is indeed another benefit.
Maybe confusing for new users that "the global state" refers to the state _inside_ the module?

Does using a module _reduce_ the need for restarting Julia, or (virtually) eliminate it?

I'll tag the original author of this section (https://github.com/tfiers/julia/commit/4b51aef), @twadleigh, to ask them what they think.

Reply:
Actually this edit changes what I meant: I think that in an "efficient interactive workflow", which this guide [is about][1], the main advantage of a module is not having to restart Julia when _editing_ structs or _removing_ methods.

Not seeing definitions outside the module seems like more of a software engineering advantage (and might even be a disadvantage for the interactive user here).
I am all about encapsulation and not being bitten by hidden state; but I'm not sure if it's worth bringing up here in this "basic editor/REPL workflow"

[1]: https://github.com/JuliaLang/julia/blob/4b51aef2a440a701608fb42fd3189bed3b2611d3/doc/manual/workflow-tips.rst?plain=1#L13